extension/src/utils/aria.ts

```diff
@@ -17,32 +24,92 @@ function bySpecificity(a: IntrinsicRole, b: IntrinsicRole) {
   return a.specificity - b.specificity
 }
 
+type AttributePredicate = (element: Element) => boolean
+
+function getConstraintPredicate(
```

Comment:
The existing implementation couldn't detect that `input` elements without a `type` attribute have the `textbox` role, so I rewrote it to something more robust.

Having multiple values comes into play when I add support for `select` elements.